core/lib/spree/testing_support/event_helpers.rb

```diff
@@ -0,0 +1,139 @@
+      # method invocations on it will be spied but not performed.
+      #
+      # Internally, it stubs {Spree::Event} to a class spy of itself.
+      #
+      # After you call this method, probably you'll want to call some of the
+      # matchers defined in this module.
+      def stub_events
+        stub_const('Spree::Event', class_spy(Spree::Event))
+      end
+
+      # @!method have_been_fired(event_name)
+      #   Matcher to test that an event has been fired via {Spree::Event#fire}
+      #
+      #   Before using this matcher, you need to call {#stub_events}.
+      #
+      #   Remember that the event listeners won't be performed.
+      #
+      #   @example
+      #     it 'fires foo event' do
+      #       stub_events
+      #
+      #       Spree::Event.fire 'foo'
+      #
+      #       expect('foo').to have_been_fired
+      #     end
+      #
+      #   It can be chain through `with` to match with the published payload:
+      #
+      #   @example
+      #     it 'fires foo event with the expected payload' do
+      #       stub_events
+      #
+      #       Spree::Event.fire 'foo', bar: :baz, qux: :quux
+      #
+      #       expect('foo').to have_been_fired.with(a_hash_including(bar: :baz))
+      #     end
+      #
+      #   @param [String, Symbol] event_name
+      matcher :have_been_fired do
+        chain :with, :payload
+
+        match do |expected_event|
+          expected_event = normalize_name(expected_event)
+          arguments = payload ? [expected_event, payload] : [expected_event, any_args]
+          expect(Spree::Event).to have_received(:fire).with(*arguments)
+        end
+
+        failure_message do |expected_event|
+          <<~MSG
+            expected #{expected_event.inspect} to have been fired.
+            Make sure that provided payload, if any, also matches.
+          MSG
+        end
+
+        def normalize_name(event_name)
+          if event_name.is_a?(String)
+            eq(event_name).or(eq(event_name.to_sym))
+          elsif event_name.is_a?(Symbol)
+            eq(event_name).or(eq(event_name.to_s))
+          else
+            raise ArgumentError, <<~MSG
+              "#{event_name.inspect} is not a valid event name. It must be a String or a Symbol."
+            MSG
+          end
+        end
+      end
+
+      # @!method have_subscription(event_name)
+      #   Matcher to test that an event got a subscription via
+      #   {Spree::Event#subscribe}
+      #
+      #   Before using this matcher, you need to call {#stub_events}.
+      #
+      #   Remember that the event listeners won't be performed.
+      #
+      #   @example
+      #     it 'fires foo event' do
+      #       stub_events
+      #
+      #       Spree::Event.subscribe 'foo'
+      #
+      #       expect('foo').to have_subscription
+      #     end
+      #   @param [String, Symbol] event_name
+      matcher :have_subscription do
```

Comment:
I'm not completely sure, but I suspect this might not work properly in a real app, where the event subscribers are subscribed during boot (rather than after the `stub_events` call). Or have you already checked?

Reply:
Yeah, you're right. It's useless for module-based subscribers. My idea was to use it for subscriptions made through the raw `Spree::Event.subscribe` method. But maybe it would be something confusing for users. Do you think we should remove it altogether?

Reply:
@waiting-for-dev I think the standard will be module-based subscriber, so yeah, I don't think it adds a lot of value right now. 😢

Reply:
:scissors: :scissors: done! The less code to maintain, the best! :slightly_smiling_face: I forced a push removing `have_subsription` and updated the commit message.